UI/src/view/navigationpanel/searchBox.js

```diff
@@ -11,80 +11,156 @@
+        }
+      });
+
+      moveButton.on('click', function() {
+        emptyResultSection();
+        moveToLocation();
+      });
+
+      clearButton.on('click', function() {
+        searchResults.empty();
+        resultsSection.hide();
+        clearSection.hide();
+      });
 
-        searchResults.html(resultItems);
-        resultsSection.show();
-        clearSection.show();
+      eventbus.on('associationNames:fetched', function(result) {
+        enableAutoComplete(coordinatesText,result);
+      });
+
+      eventbus.on('associationNames:reload', function() {
+        locationSearch.privateRoadAssociationNames();
+      });
+    };
+
+    var moveToLocation = function() {
+      resultsSection.show();
+      jQuery('#search-results').append('<div class="spinner-overlay-search"></div>');
+
+      var showDialog = function(message) {
+        instructionsPopup.show(message, 3000);
       };
-      var moveToLocation = function() {
-        var showDialog = function(message) {
-          instructionsPopup.show(message, 3000);
-        };
-        locationSearch.search(coordinatesText.val()).then(function(results) {
-          populateSearchResults(results);
-          if (results.length === 1) {
-            var result = results[0];
-            if (result.resultType.indexOf("Mtstop")>-1) {
-              window.location.hash = "#massTransitStop/";
-              window.location.hash="#massTransitStop/"+result.nationalId;
-            }
-            else if (result.resultType.indexOf("Link-id")>-1) {
+      locationSearch.search(coordinatesText.val()).then(function(results) {
+        populateSearchResults(results);
+        if (results.length === 1) {
+          var result = results[0];
+          if (result.resultType.indexOf("Mtstop")>-1) {
+            window.location.hash = "#massTransitStop/";
+            window.location.hash="#massTransitStop/"+result.nationalId;
+          }
+          else if (result.resultType.indexOf("Link-id")>-1) {
+            window.location.hash = "#linkProperty/";
+            eventbus.trigger('coordinates:selected', {lon: result.lon, lat: result.lat});
+            window.location.hash = "#linkProperty/" + coordinatesText.val();
+          } else if (result.resultType.indexOf("SpeedLimit")>-1) {
+            eventbus.once('layer:speedLimit:moved', function() {
+              eventbus.trigger('speedLimit:selectByLinkId', result.linkid);
+            });
+            eventbus.trigger('coordinates:selected', { lon: result.lon, lat: result.lat });
+          }
+          else {
+            eventbus.trigger('coordinates:selected', { lon: result.lon, lat: result.lat });
+          }
+        }
+      }).fail(showDialog);
+    };
+
+    var populateSearchResults = function(results) {
+      var resultItems = _.chain(results)
+        .sortBy('distance')
+        .map(function(result) {
+
+          var shownText = function(value){
+            if(value.resultType.indexOf("association")>-1)
```

Comment:
would it be possible to on locationSearch.js construct the title in this format? in this way we could avoid this all together